Bottom: dbe25949bbbf2ce84e889234850e484e34abf082
Top:    b8083e94852eec1a1488e8b44683f3a9bc82d884
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-09 23:08:11 +0000

WORK IN PROGRESS


---

diff --git a/src/hid/common/hidnogui.c b/src/hid/common/hidnogui.c
index 2b98a75..f8ce9df 100644
--- a/src/hid/common/hidnogui.c
+++ b/src/hid/common/hidnogui.c
@@ -294,13 +294,19 @@ static char *
 nogui_prompt_for (const char *msg, const char *default_string)
 {
   static char buf[1024];
+  char *s;
+
   if (default_string)
     printf ("%s [%s] : ", msg, default_string);
   else
     printf ("%s : ", msg);
-  fgets (buf, 1024, stdin);
-  if (buf[0] == 0 && default_string)
-    strcpy (buf, default_string);
+
+  s = fgets (buf, 1024, stdin);
+  if (default_string && (s == NULL || buf[0] == 0))
+    {
+      strncpy (buf, default_string);
+    }
+
   return buf;
 }
